src/safeposix/syscalls/net_calls.rs

```diff
@@ -1358,39 +1490,149 @@ impl Cage {
+    ///   have the same value, so a portable application should check for both
+    ///   possibilities.
+    ///
+    /// ** EAGAIN - (Internet domain datagram sockets) The socket referred to by
+    ///   sockfd had not previously been bound to an address and, upon
+    ///   attempting to bind it to an ephemeral port, it was determined that all
+    ///   port numbers in the ephemeral port range are currently in use.  See
+    ///   the discussion of /proc/sys/net/ipv4/ip_local_port_range in ip(7).
+    ///
+    /// * EALREADY - Another Fast Open is in progress.
+    ///
+    /// ** EBADF - sockfd is not a valid open file descriptor.
+    ///
+    /// * ECONNRESET - Connection reset by peer.
+    ///
+    /// * EDESTADDRREQ - The socket is not connection-mode, and no peer address
+    ///   is set.
+    ///
+    /// * EFAULT - An invalid user space address was specified for an argument.
+    ///
+    /// * EINTR - A signal occurred before any data was transmitted; see
+    ///   signal(7).
+    ///
+    /// ** EINVAL - Invalid argument passed.
+    ///
+    /// * EISCONN - The connection-mode socket was connected already but a
+    ///   recipient was specified.  (Now either this error is returned, or the
+    ///   recipient specification is ignored.)
+    ///
+    /// * EMSGSIZE - The socket type requires that message be sent atomically,
+    ///   and the size of the message to be sent made this impossible.
+    ///
+    /// * ENOBUFS - The output queue for a network interface was full.  This
+    ///   generally indicates that the interface has stopped sending, but may be
+    ///   caused by transient congestion. (Normally, this does not occur in
+    ///   Linux.  Packets are just silently dropped when a device queue
+    ///   overflows.)
+    ///
+    /// * ENOMEM - No memory available.
+    ///
+    /// ** ENOTCONN - The socket is not connected, and no target has been given.
+    ///
+    /// ** ENOTSOCK - The file descriptor sockfd does not refer to a socket.
+    ///
+    /// ** EOPNOTSUPP - Some bit in the flags argument is inappropriate for the
+    ///   socket type.
+    ///
+    /// ** EPIPE - The local end has been shut down on a connection oriented
+    ///   socket.  In this case, the process will also receive a SIGPIPE unless
+    ///   MSG_NOSIGNAL is set.
+    ///
+    /// ** Indicates the error may be returned from RustPOSIX
+    ///
+    /// ### Panics:
+    ///
+    /// * invalid or out-of-bounds file descriptor, calling unwrap() on it will
+    ///   cause a panic.
+    /// * Unknown errno value from sendto returned, will cause panic.
+    ///
+    /// for more detailed description of all the commands and return values, see
+    /// [send(2)](https://linux.die.net/man/2/send)
     pub fn send_syscall(&self, fd: i32, buf: *const u8, buflen: usize, flags: i32) -> i32 {
+        //BUG:
+        //If fd is out of range of [0,MAXFD], process will panic
```

Comment:
This is not good.  Should return an error instead of panicing

Reply:
[This is a bug throughout](https://github.com/Lind-Project/safeposix-rust/issues/263) and is going to be updated globally. Should have someone fixing this next week.